scripting/internal/codec/decoders.go

```
@@ -28,6 +28,26 @@ func DecodeString[T any](s js.Scope[T], v js.Value[T]) (string, error) {
+func DecodeBinaryString[T any](s js.Scope[T], v js.Value[T]) ([]byte, error) {
+	if v == nil {
+		v = s.Undefined()
+	}
+	str := v.String()
+	bytes := make([]byte, 0, len(str))
+	for _, r := range str {
+		if r > 0xFF {
+			return nil, fmt.Errorf(
+				"codec: binary string contains a character outside the Latin-1 range")
+		}
+		bytes = append(bytes, byte(r))
```

golangci-lint (2.12.2)

[high] G115: integer overflow conversion rune -> byte

(gosec)